app/assets/javascripts/polaris_view_components/tooltip_controller.js

```diff
@@ -0,0 +1,73 @@
+import { Controller } from "@hotwired/stimulus";
+import { createPopper } from "@popperjs/core/dist/esm";
+
+export default class extends Controller {
+  static targets = ["template"];
+  static values = { active: Boolean, position: String }
+
+  initialize() {
+    this.shownTooltip = null;
+  }
+
+  getOffset() {
+    switch (this.positionValue) {
+      case "top", "bottom":
+        return [0, 4]
+      case "left", "right":
+        return [0, 6]
+      default:
+        return [0, 4]
+    }
+  }
+
+  show(event) {
+    if (!this.activeValue) return;
+
+    const popperOptions = {
+      placement: this.positionValue,
+      modifiers: [
+        {
+          name: "matchReferenceSize",
```

Comment:
This modifier causes huge performance issues. The page freezes in Safari and Firefox:
![CleanShot 2023-07-18 at 16 46 43@2x](https://github.com/baoagency/polaris_view_components/assets/839922/842ba517-3934-4e78-92eb-1726f4220c3b)

And when turned off everything seems to work fine. Maybe just remove it?

Reply:
Just double checked it and the tooltip looks the same without this modifier. I took this code from here https://github.com/floating-ui/floating-ui/issues/794 but I think it's irrelevant now